src/test/java/com/iexec/core/sms/SmsServiceTests.java

```diff
@@ -56,4 +59,15 @@ void shouldGetEmptyAddressForStandardTask() {
                 .isEqualTo(BytesUtils.EMPTY_ADDRESS);
         verify(smsClient, never()).generateTeeChallenge(anyString());
     }
+
```

Comment:
If you have time you can add `shouldNotGetEnclaveChallengeForTeeTaskWhenEmptySmsResponse` and/or check replace implem check to something like `if(StringUtils.isEmpty(teeChallengePublicKey))`.